oauthenticator/multiauthenticator.py

```diff
@@ -0,0 +1,124 @@
+"""
+Custom Authenticator to use multiple OAuth providers with JupyterHub
+
+Example of configuration:
+
+    from oauthenticator.github import GitHubOAuthenticator
+    from oauthenticator.google import GoogleOAuthenticator
+
+    c.MultiAuthenticator.authenticators = [
+        (GitHubOAuthenticator, '/github', {
+            'client_id': 'xxxx',
+            'client_secret': 'xxxx',
+            'oauth_callback_url': 'http://example.com/hub/github/oauth_callback'
+        }),
+        (GoogleOAuthenticator, '/google', {
+            'client_id': 'xxxx',
+            'client_secret': 'xxxx',
+            'oauth_callback_url': 'http://example.com/hub/google/oauth_callback'
+        }),
+        (PAMAuthenticator, "/pam", {"service_name": "PAM"}),
+    ]
+
+    c.JupyterHub.authenticator_class = 'oauthenticator.MultiAuthenticator.MultiAuthenticator'
+
+The same Authenticator class can be used several to support different providers.
+
+"""
+from jupyterhub.auth import Authenticator
+from jupyterhub.utils import url_path_join
+from traitlets import List
+
+
+class URLScopeMixin(object):
+    """Mixin class that adds the"""
+
+    scope = ""
```

Comment:
Should be renamed to avoid colision with `scope` variable defined in `oauth2`, like this, it breaks with gitlab.
```suggestion
    url_scope = ""
```